enterprise/internal/campaigns/reconciler.go

```diff
@@ -555,6 +574,77 @@ func loadExternalService(ctx context.Context, reposStore repos.Store, repo *repo
 	return externalService, nil
 }
 
+func loadCampaign(ctx context.Context, tx *Store, id int64) (*campaigns.Campaign, error) {
+	if id == 0 {
+		return nil, errors.New("changeset has no owning campaign")
+	}
+
+	campaign, err := tx.GetCampaign(ctx, GetCampaignOpts{ID: id})
+	if err != nil {
+		return nil, errors.Wrapf(err, "retrieving owning campaign: %d", id)
+	} else if campaign == nil {
+		return nil, errors.Errorf("campaign not found: %d", id)
+	}
```

Comment:
I think this doesn't work, we throw ErrNoResults errors

```suggestion
	campaign, err := tx.GetCampaign(ctx, GetCampaignOpts{ID: id})
	if err != nil && err != ErrNoResults {
		return nil, errors.Wrapf(err, "retrieving owning campaign: %d", id)
	} else if campaign == nil {
		return nil, errors.Errorf("campaign not found: %d", id)
	}
```